jobbergate-api/jobbergate_api/file_validation.py

```diff
@@ -0,0 +1,265 @@
+    """
+    Use this decorator to register functions to check the uploaded files.
+
+    It creates a new entry on ``check_uploaded_files_dispatch``.
+
+    :return CheckUploadedFilesEquation: The decorated function.
+    """
+    check_uploaded_files_dispatch.append(checker)
+
+    @wraps(checker)
+    def decorator(file_list: List[UploadFile]) -> None:
+        return checker(file_list)
+
+    return decorator
+
+
+@register_check_uploaded_files
+def check_uploaded_files_extensions(file_list: List[UploadFile]) -> None:
+    """
+    Check the list of uploaded files to count the file extensions.
+
+    Raise UploadedFilesValidationError if the business roles are not met.
+
+    :param List[UploadFile] file_list: Upload file list.
+    """
+    extension_counter = Counter(get_suffix(f) for f in file_list)
+    logger.debug(f"Checking uploaded files extensions: {extension_counter=}")
+    with UploadedFilesValidationError.check_expressions("Invalid file extensions") as check:
+        check(
+            extension_counter[".py"] == 1,
+            "one application source file (.py)",
+        )
+        check(
+            0 <= extension_counter[".yaml"] <= 1,
+            "one (optional) application config file (.yaml)",
+        )
+        check(
+            extension_counter[".j2"] + extension_counter[".jinja2"] >= 1,
+            "one or more template files (.j2 and/or .jinja2)",
+        )
+
+
+@register_check_uploaded_files
+def check_uploaded_files_content_type(file_list: List[UploadFile]) -> None:
+    """
+    Check the list of uploaded files to confirm they are specified as plain text.
+
+    Raise UploadedFilesValidationError if the business roles are not met.
+
+    :param List[UploadFile] file_list: Upload file list.
+    """
+    logger.debug("Checking uploaded files content types")
+    expected = "text/plain"
+    extension_list = ", ".join(syntax_validation_dispatch.keys())
+    UploadedFilesValidationError.require_condition(
+        all(f.content_type == expected for f in file_list if get_suffix(f) in syntax_validation_dispatch),
+        f"The content of the files {extension_list} is expected to be {expected}",
+    )
+
+
+@register_check_uploaded_files
+def check_uploaded_files_syntax(file_list: List[UploadFile]) -> None:
+    """
+    Check the list of uploaded files to verify their syntax.
+
+    Raise UploadedFilesValidationError if the business roles are not met.
+
+    :param List[UploadFile] file_list: Upload file list.
+    """
+    logger.debug("Checking uploaded files, validating source code")
+
+    with UploadedFilesValidationError.check_expressions("Invalid syntax on uploaded file(s)") as check:
+        for f in file_list:
+            suffix = get_suffix(f)
+            if suffix not in syntax_validation_dispatch:
+                continue
+            check(syntax_validation_dispatch[suffix](f.file.read()), f.filename)
+            f.file.seek(0)
```

Comment:
I need to test it because there is no `open()`, I just used `seek(0)` to send the cursor back to the start of the file in case another read operation is needed.